docs/interactions.rst

```diff
@@ -0,0 +1,274 @@
+-------------------------
+Pincer provides an API for all three interaction command types. The only thing that varies is the function signiture.
+
+.. code-block:: python
+
+    from pincer.commands import command, user_command, message_command
+    from pincer.objects import MessageContext, UserMessage, User
+    ...
+
+    @command
+    # Can have any amount of inputs
+    async def ping(self, ctx: MessageContext, arg1: str, arg2: str):
+        return "pong"
+
+    # Must have a parameter for users. User can be a GuildMember. All User
+    # methods are available to GuildMember because GuildMember inherits from
+    # User.
+    async def user_ping(self, ctx: MessageContext, user: User):
+        return "pong"
+
+    # Must have a parameter for messages.
+    async def message_ping(self, ctx: MessageContext, message: UserMessage):
+        return "pong"
+
+Interaction Timeout
+-------------------
+Interactions time out after 3 seconds. To extend the timeout to 15 minutes you can run :meth:`ack` from
+:class:`~pincer.objects.MessageContext`. :class:`~pincer.objects.app.interaction_flags.InteractionFlags` is available in this method.
+
+Arguments
+---------
+
+Arguments are more variables in the command. Notice how `word` is typehinted as string.
+Pincer uses type hints to infer the argument type that you want.
+
+.. code-block:: python
+
+    @command
+    async def say(self, word: str):
+        return word
+
+The list of possible type hints is as follows:
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Class
+     - Command Argument Type
+   * - :class:`str`
+     - String
+   * - :class:`int`
+     - Integer
+   * - :class:`bool`
+     - Boolean
+   * - :class:`float`
+     - Number
+   * - :class:`pincer.objects.User`
+     - User
+   * - :class:`pincer.objects.Channel`
+     - Channel
+   * - :class:`pincer.objects.Role`
+     - Role
+   * - :class:`pincer.objects.Mentionable`
+     - Mentionable
+
+You might want to specify more information for your arguments. If you want a description for your command you will have to use the
+:class:`~pincer.commands.arg_types.Description` type. Modifier types like this need to be inside of the :class:`~pincer.commands.arg_types.CommandArg`
+type.
+
+.. code-block:: python
+
+    from pincer.commands import CommandArg, Description
+    from pincer.objects import MessageContext
+
+    @command
+    async def say(self,
+        ctx: MessageContext,
+        word: CommandArg[
+          str,
+          # This will likely marked as incorrect by your linter but it is
+          # valid python. Simply append # type: ignore for most linters and
+          # noqa: F722 if you are using Flake8.
+          Description["A word that the bot will say."]  # type: ignore # noqa: F722
+        ]
+    ):
+        # Returns the name of the user that initiated the interaction
+        return word
+
+Arguments will be an optional argument in Discord if they are an optional argument in Python.
```

Comment:
Same here, struggling to understand the beginning of the sentence lol

Reply:
Changed to "Parameters will be an optional slash command arguments if they have a default value in Python." and added an example.